src/SettingsView/SettingsView.js

```diff
@@ -14,7 +14,28 @@ import IconAdd from 'material-ui/svg-icons/content/add';
 import config from './../config/config';
 
 class SettingsView extends ConnectionComponent {
+    constructor() {
+        super();
+        this.state = {
+            searchText: ''
+        };
+        this.handleSearchInput = this.handleSearchInput.bind(this);
+        this.searchMatches = this.searchMatches.bind(this);
+    }
 
+    handleSearchInput(value) {
+        this.setState({
+            searchText: value
+        });
+    }
+
+    searchMatches(eventType, searchText) {
+        if (!searchText || !searchText.value) {

```

Comment:
Because I was still in Java with my mind and that would have thrown a Nullpointer Exception there. ;)